### DISCUSSION TITLE
Relation or JSON Object Attributes

### DISCUSSION THREAD

**Discussion by ParisZX** [appwrite/appwrite#4816] [Q&A] [answered]
Hi!

I wanted to ask if there is a plan in the future to add new attribute options to the databases, like a relation/foreign key type or a JSON object type of attribute.

I searched through the docs and the discussions here, and found this issue for example: #2901 but it didn't have an anwer. My understanding is that this is not supported at the time, but will it be added at some point?

**Comment by stnguyen90** [comment, marked as answer]
@ParisZX, for now, you can manually use a string attribute for the related document's ID. For example, check out [this](https://levelup.gitconnected.com/building-a-location-based-app-with-appwrite-48a2e2b6d4c2#a658) design.

For additional relationship support, follow:

* https://github.com/appwrite/appwrite/issues/2735

**Reply by ParisZX** [reply]
Thanks for all the information!

**Comment by sneakylenny** [comment]
To update on this @eldadfux notified in [#2735](https://github.com/appwrite/appwrite/issues/2735#issuecomment-1438913308) that they're recently started working on it. 👷

I wonder if there are already some ideas about how it's going to work!
Will it be like traditional table references, like in SQL? Will it be like nested documents, like in MongoDB? Or will it be like SurrealDB where they store collection name and item ID?

What do you guys think?
